EIPS/eip-7336.md

```diff
@@ -0,0 +1,104 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-07-14
+requires: 165
+---
+
+## Abstract
+
+The objective of this EIP is to propose the **Universal Domain Namespace(UDoN)**, which aims to facilitate onchain registry entries. Each entry in this structure adopts a mapping type, allowing for interactive capabilities across multiple registries like [ENS](./eip-137.md) and [NFT accounts](./eip-6551.md).
+
+## Motivation
+
+The necessity for a unified method of data retrieval across various registries is apparent. A structured methodology that allows for seamless reading of data from one registry to another, effectively linking chains of information, is thus proposed. This proposal seeks to standardize an interface that not only supports interaction but also promotes integration with multi-protocol registries. With the utilization of such an interface, a single deployed contract can retrieve records from different combinations of registries, such as ENS, the registry for royalty payments of NFTs, the registry for smart contract accounts owned by NFT tokens, and other registries as defined by varying protocols.
+
+## Specification
+
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+**The Universal Domain Namespace(UDoN)** is a singular contract that enables the retrieval of data from multiple registries, each defined under different protocols within its own namespace. These registries employ distinct methods for resolving the key-value pairs.
+
+Resolving a node in **UDoN** involves a two-step process. Firstly, the **UDoN** is invoked with the protocol specified by the caller to resolve their namespace. This call returns the corresponding registry associated with that protocol. Secondly, using the key that needs resolution, if the record exists in the returned registry, the registry itself retrieves the associated value using its own implementation. This retrieval process can involve directly returning the value or utilizing a resolver.
+
+### Typical Flow
+
+![Typical Flow](../assets/eip-7336/workflow.png)
+
+### Interface
+
+A new UDoN interface is defined, consisting of the following method:
+
+```solidity
+interface IERC7336 {
+  struct Result {
+      address   resAddr;
+      string    resText;
+      bytes32   resAbi;
+      bytes32   resPublickey;
+  }
+
+  /// @notice Returns the multi type data by retrieving value from different registries
+  /// @param registryAddress An address of target registry
+  /// @param node A key to lookup data for
+  /// @return The value is retrieved from specific registry
+  function getValue(address registryAddress, byte32 node) external view returns (Result res);
+}
+```
+
+Smart contracts implementing the standard MUST implement all of the functions in the interface.
+
```

Comment:
I don't believe the Specification section is detailed enough to allow an implementer to implement this proposal. There should be enough information here that someone reading this proposal can implement the standard without necessarily having to read the reference implementation.